server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java

```diff
@@ -116,6 +117,28 @@ public ReadOnlyEngine(EngineConfig config, SeqNoStats seqNoStats, TranslogStats
         }
     }
 
+    protected DirectoryReader open(final Directory directory) throws IOException {
```

Comment:
What is the reason for the addition of this protected method in this PR?

Reply:
it will be needed in @tlrx work and I have some work based on it so I just tried to move all changes from  #33903 on ReadOnlyEngine to this pr

Reply:
Okay, that’s fine then.